streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -554,6 +555,36 @@ protected void maybeClean() {
         }
     }
 
+
+    /**
+     * @param assignment The Topics and partitions assigned to this stream thread
+     */
+    protected void maybeSetConsumerOffsets(Collection<TopicPartition> assignment) {
```

Comment:
I talked to @guozhangwang about this PR today and while we discussed, it came to my mind that this part is actually no correct. We want to set different auto.resets policies but we define starting offsets here. The difference is, that auto.offset setting should only have any effect, if there are no valid committed offsets found. But this part does `seekToBeginning()` and `seekToEnd()` in any case and thus overwriting valid committed offsets, too.

To fix this, the client's reset policy must actually set to `none` and in `consumer.poll()` we need to catch `InvalidOffsetException` and do the seek according to the specified reset strategy (either topic configured ot the global one) if we catch thus an exception.

Does this make sense?